pkg/volume/util/operationexecutor/node_expander.go

```diff
@@ -37,7 +37,7 @@ type NodeExpander struct {
 	// computed via precheck
 	pvcStatusCap resource.Quantity
 	pvCap        resource.Quantity
-	resizeStatus *v1.PersistentVolumeClaimResizeStatus
+	resizeStatus *v1.ClaimResourceStatus
```

Comment:
Does this have to be a pointer?

Reply:
So, it doesn't have to be a pointer but I am slightly uncomfortable with the fact that, we have defined `type ClaimResourceStatus string` as a Enum, but there is no *possible* value for that enum, which results in empty string. At least not in the API.

But as a side-effect of golang's treatment of empty and nil maps mostly being the same, we will end up reading empty string as a value of `resizeStatus`.  But according to the API, that is not a *valid* state (strictly speaking).

So I am using pointer here, to stay as close as possible to the API.  I can either ignore the inconsistency within the API (a language that strongly checks against possible enum values will complain) or I can add an empty string as a possible value of `ClaimResourceStatus`.

Reply:
Changed it to non-pointer.